src/Language/Mimsa/Interpreter.hs

```diff
@@ -43,20 +44,27 @@ useVarFromScope name = do
       scope' <- get
       throwError $ CouldNotFindVar scope' name
 
-wrappedName :: Name -> Name
-wrappedName (Name n) = Name (n <> "__unwrapped")
+wrappedName :: Variable -> Variable
+wrappedName = appendToVar "__unwrapped"
 
-wrappedVarName :: Name -> Int -> Name
-wrappedVarName name i = Name $ wrapped' <> (T.pack $ show i)
+appendToVar :: Text -> Variable -> Variable
+appendToVar tx (NamedVar (Name n)) = NamedVar (Name (n <> tx))
+appendToVar tx (BuiltIn (Name n)) = BuiltIn (Name (n <> tx))
+appendToVar _ a = a
+
+wrappedVarName :: Variable -> Int -> Variable
+wrappedVarName name i = appendToVar label name
   where
-    (Name wrapped') = wrappedName name
+    label = "__unwrapped" <> T.pack (show i)
 
-unwrap :: Name -> Maybe Name
-unwrap (Name n) = Name <$> T.stripSuffix "__unwrapped" n
+unwrap :: Variable -> Maybe Variable
+unwrap (NamedVar (Name n)) = NamedVar . Name <$> T.stripSuffix "__unwrapped" n
+unwrap (BuiltIn (Name n)) = BuiltIn . Name <$> T.stripSuffix "__unwrapped" n
+unwrap a = Just a
 
 -- return lambda to new function, with __unwrapped in the name
 -- if Name ends in __unwrapped then run it else return new function...
-useVarFromBuiltIn :: Name -> App Expr
+useVarFromBuiltIn :: Variable -> App (Expr Variable)
```

Comment:
I'm sure now we are using `Variable` instead of just strings, that we can do this a lot more nicely, but now CBA.